### PR TITLE
Added (missing) KWS models.

### PR DESCRIPTION
**Pull request by GuilhermeMota93**
## JIRA
Link of the JIRA tickets the PR relates to. 
[KWS-1972](https://jira.superawesome.tv/browse/KWS-1972)

## DESCRIPTION
Added missing KWS models.

## SCREENSHOTS
Attach any screenshot or .gif file to show how the PR fixes an issue or the new behaviour introduced by it.
`n/a`

### CODE REVIEW

**Comment by devgabrielcoman** [comment]
Pass all the tests!!!!

File: Example/Tests/AddressModelsProtocolsTests.swift [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        
+        let testClass = MockClassAddressModelProtocol.init(withStreet: street, city: city, postCode: postCode, country: country)
+        
+        expect(testClass).toNot(beNil())
+        expect(testClass.street).to(equal(street))
+        expect(testClass.city).to(equal(city))
+        expect(testClass.postCode).to(equal(postCode))
+        expect(testClass.country).to(equal(country))
+    }
+
```
Comment (Tbruni85) [inline Example/Tests/AddressModelsProtocolsTests.swift]: no space pls

File: Example/Tests/AppDataModelProtocolsTests.swift [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        expect(testClass.results[1].value).to(equal(value2))
+        
+        expect(testClass.results[2].name).to(equal(name3))
+        expect(testClass.results[2].value).to(equal(value3))
+        
+        expect(testClass.count).to(equal(count))
+        expect(testClass.offset).to(equal(offset))
+        expect(testClass.limit).to(equal(limit))
+    }
+
```
Comment (Tbruni85) [inline Example/Tests/AppDataModelProtocolsTests.swift]: no space pls

File: Example/Tests/PermissionsModelsProtocolsTests.swift [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        expect(testClass.email).to(beNil())
+        expect(testClass.streetAddress).to(beNil())
+        expect(testClass.city).to(beNil())
+        expect(testClass.postalCode).to(beNil())
+        expect(testClass.country).to(beNil())
+        expect(testClass.newsletter).to(beNil())
+        expect(testClass.competition).to(beNil())
+    }
+}
+
```
Comment (Tbruni85) [inline Example/Tests/PermissionsModelsProtocolsTests.swift]: no space pls

File: Example/Tests/ScoringModelsProtocolsTests.swift [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+        
+        let hasTriggeredEvent = true
+        
+        let testClass = MockClassHasTriggeredEventModelProtocol.init(hasTriggeredEvent: hasTriggeredEvent)
+        
+        expect(testClass).toNot(beNil())
+        expect(testClass.hasTriggeredEvent).to(equal(hasTriggeredEvent))
+        
+    }
+
```
Comment (Tbruni85) [inline Example/Tests/ScoringModelsProtocolsTests.swift]: no space pls

File: Example/Tests/ScoringModelsProtocolsTests.swift [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+    // MARK: HasTriggeredEventModelProtocol
+    func testCorrectClassInit_conformed_to_HasTriggeredEventModelProtocol() {
+        
+        let hasTriggeredEvent = true
+        
+        let testClass = MockClassHasTriggeredEventModelProtocol.init(hasTriggeredEvent: hasTriggeredEvent)
+        
+        expect(testClass).toNot(beNil())
+        expect(testClass.hasTriggeredEvent).to(equal(hasTriggeredEvent))
+
```
Comment (Tbruni85) [inline Example/Tests/ScoringModelsProtocolsTests.swift]: no space pls

File: Example/Tests/UserModelsProtocolsTests.swift [hunk 6]
```diff
@@ -106,11 +107,90 @@ class UserModelsProtocolsTests: XCTestCase {
...
+        let testClass = MockClassAppProfileModelProtocol.init(customField1: customField1, customField2: customField2, avatarId: avatarId, name: name)
+        
+        expect(testClass).toNot(beNil())
+        expect(testClass.customField1).to(equal(customField1))
+        expect(testClass.customField2).to(equal(customField2))
+        expect(testClass.avatarId).to(equal(avatarId))
+        expect(testClass.name).to(equal(name))
+    }
+}
+
```
Comment (Tbruni85) [inline Example/Tests/UserModelsProtocolsTests.swift]: no space pls

File: SAProtobufs/Classes/Models Protocols/AddressModelProtocol.swift [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+import Foundation
+
+public protocol AddressModelProtocol: ModelProtocol {
+    
+    var street: String? { get set }
+    var city: String? { get set }
+    var postCode: String? { get set }
+    var country: String? { get set }
+
```
Comment (Tbruni85) [inline SAProtobufs/Classes/Models Protocols/AddressModelProtocol.swift]: no space pls

File: SAProtobufs/Classes/Models Protocols/ScoreModelsProtocols.swift [hunk 8]
```diff
@@ -7,10 +7,36 @@
...
+public protocol LeaderWrapperModelProtocol: ModelProtocol, ListTraitProtocol  {
+    var results: [LeaderModelProtocol] { get set }
+}
+
+//KWS
+public protocol HasTriggeredEventModelProtocol: ModelProtocol {
+    var hasTriggeredEvent: Bool { get set }
+}
+
+
```
Comment (Tbruni85) [inline SAProtobufs/Classes/Models Protocols/ScoreModelsProtocols.swift]: no space pls

File: SAProtobufs/Classes/Models Protocols/UserModelsProtocols.swift [hunk 9]
```diff
@@ -37,3 +37,29 @@ public protocol UserSocialDetailsWrapperModelProtocol: ModelProtocol {
...
+    var dateOfBirth: String? { get set }
+    var gender: String? { get set }
+    var email: String? { get set }
+    var hasSetParentEmail: Bool? { get set }
+    var createdAt: String? { get set }
+    var address: AddressModelProtocol? { get set }
+    var applicationProfile: AppProfileModelProtocol? { get set }
+    var applicationPermissions: PermissionsModelProtocols? { get set }
+    var points: PointsModelProtocols? { get set }
+
```
Comment (Tbruni85) [inline SAProtobufs/Classes/Models Protocols/UserModelsProtocols.swift]: no space pls

File: Example/Tests/ScoringModelsProtocolsTests.swift [hunk 10]
```diff
@@ -0,0 +1,214 @@
...
+    func testCorrectClassInit_conformed_to_HasTriggeredEventModelProtocol() {
+        
+        let hasTriggeredEvent = true
+        
+        let testClass = MockClassHasTriggeredEventModelProtocol.init(hasTriggeredEvent: hasTriggeredEvent)
+        
+        expect(testClass).toNot(beNil())
+        expect(testClass.hasTriggeredEvent).to(equal(hasTriggeredEvent))
+    }
+
```
Comment (Tbruni85) [inline Example/Tests/ScoringModelsProtocolsTests.swift]: space

File: Example/Tests/ScoringModelsProtocolsTests.swift [hunk 11]
```diff
@@ -0,0 +1,214 @@
...
+    var rank: Int
+    
+    init(score: Int, rank: Int) {
+        
+        self.score = score
+        self.rank = rank
+    }
+}
+
+
```
Comment (Tbruni85) [inline Example/Tests/ScoringModelsProtocolsTests.swift]: space

File: SAProtobufs/Classes/Models Protocols/ScoreModelsProtocols.swift [hunk 12]
```diff
@@ -7,10 +7,36 @@
...
+public protocol LeaderWrapperModelProtocol: ModelProtocol, ListTraitProtocol  {
+    var results: [LeaderModelProtocol] { get set }
+}
+
+//KWS
+public protocol HasTriggeredEventModelProtocol: ModelProtocol {
+   
+    var hasTriggeredEvent: Bool { get set }
+}
+
```
Comment (Tbruni85) [inline SAProtobufs/Classes/Models Protocols/ScoreModelsProtocols.swift]: spaces

**Comment by Tbruni85** [comment]
👍

**Pull request merged**